packages/ripple-binary-codec/src/binary.ts

```diff
@@ -16,26 +20,43 @@ import * as bigInt from 'big-integer'
  * Construct a BinaryParser
  *
  * @param bytes hex-string to construct BinaryParser from
+ * @param definitions rippled definitions used to parse the values of transaction types and such.
+ *                          Can be customized for sidechains and amendments.
  * @returns A BinaryParser
  */
-const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes)
+const makeParser = (
+  bytes: string,
+  definitions?: XrplDefinitionsBase,
```

Comment:
Does this not need `definitions?: XrplDefinitionsBase = undefined`?

Reply:
ditto for all the similar methods

Reply:
Isn't that implied by the optional parameter?

Reply:
Why is it undefined in some places and equal to DEFAULT_DEFINITIONS in others? It doesn't seem super uniform where it's included and where it's not.

Reply:
I only included DEFAULT_DEFINITIONS in places where the underlying functions didn't know how to interpret undefined. In cases where the sub-functions had their own default, I just passed in undefined to allow the lower levels to specify details rather than enforcing the default of a higher function. 

This specifically matters so that an update to the binary-codec can improve upon the default without requiring an update to xrpl.js (Although I suppose you may still need an updated version in order to have the binary-codec dependency be updated)

Ex. `encode` (undefined) -> `serializeObject` (undefined) -> `.from` (lowest level, so defaults to `DEFAULT_DEFINITIONS`)